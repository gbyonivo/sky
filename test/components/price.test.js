import React from 'react';
import { createStore, applyMiddleware } from 'redux';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import combinedReducer from '../../src/reducers';
import Price from '../../src/components/price';

const priceProps = {
  price: { num: 1, den: 2, decimal: 0.5 },
  status: { suspended: true, active: false },
  suffixText: 'Sky',
  className: 'blue',
  type: 'short',
  isDecimalFormat: true
};

describe('Price component', () => {
  it('renders correctly', () => {
    const store = createStore(combinedReducer(), { footballEventsReducer: { isDecimalFormat: false, events: [] } });
    const actual = renderer.create(<Provider store={store}><Price {...priceProps}/></Provider>);
    expect(actual).toMatchSnapshot();
  });
});