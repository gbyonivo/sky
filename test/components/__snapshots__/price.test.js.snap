// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price component renders correctly 1`] = `
<div
  className="
    price
    short
    blue
    suspended"
>
  <span
    className="text"
  >
    Sky
  </span>
  <span
    className="priceValue"
  >
    -
  </span>
</div>
`;
